src/routes/mint/[address]/[id].svelte

```diff
@@ -1,15 +1,141 @@
 	import { onMount } from 'svelte';
-
+	import { env } from '$lib/constants';
+	// import { tokenUriToMetaData } from '$lib/nft-utils';
+	// import { evaluateNft, getTokenUriType, getImageType } from '$lib/utils/functions';
+	// import defaultAbi from '$lib/defaultAbi';
+	// import { Contract, ethers } from 'ethers';
+	import { user } from '$lib/stores/user';
+	
+	let errorPreparingMint = false;
+	let readyToMint = false;
+	let newTokenURI = '';
+	
+	let newMetadata = {};
+	
 	onMount(async () => {
 		// Do initialization things here
+		
+		// const node1 = await Ipfs.create();
+		// const result = await node1.add("hello world2");
+		// debugger;
+		
+		
+		// If you directly load this page, the wallet isn't connected so it can't get blocknumber 
+		let blockNumber = "unknown";
+		if ($user.provider) {
+			blockNumber = await $user.provider.getBlockNumber();
+		}
+		
+		// First, get the token metadata (i.e. result of calling tokenURI and fetching)
+		const metadata = await getTokenMetadata();
+		newMetadata['originalTokenMetadata'] = metadata;
+		newMetadata['description'] = `This is a Snapshot of NFT ${metadata.name} at block #${blockNumber}`;
+		
+		// Then, retrieve the image so you can put it on ipfs
+		let image_url = metadata.image;
+		// returns 'ipfs', 'http', or 'embedded', or 'other'
+		const imageType = await getImageType(image_url);
+		console.log("image type:", imageType);
+		
 		const node = await Ipfs.create();
-		debugger;
-		// const client = await create('https://ipfs.infura.io:5001/api/v0');
-		// let results = await client.add('a string that will live on IPFS');
-		console.log(results);
+		
+		// TODO: eliminate false positives due to http://gateway/ipfs
+		// They should be treated as IPFS links and may not need to be re-added to ipfs
+		if (imageType == 'http') {
+			// Get image data 
+			const response = await fetch(image_url);
+			if (!response.ok) {  // TODO: Show fail to fetch in UI 
+				console.log('Issue fetching image: ', response);
+				errorPreparingMint = true;
+				return;
+			}
+			let result;
+			try {
+				const imageData = await response.blob();
+				result = await node.add(imageData);
+				console.log("ipfs result", result);
+				
+				// if you get ipfs, overwrite the image_url
+				image_url = "ipfs://" + result.path + "/";
+			} catch (error) {
+				console.log('Error uploading image to IPFS: ', error);
+				errorPreparingMint = true;
+				return; 
+			}
+		}
+		
+		newMetadata['image_url'] = image_url;
+		newMetadata['image'] = getPolaroidVersion(image_url);
+		
+		// Now newMetadata is complete, upload it so we can use it as the tokenURI
+		try {
+			const result = await node.add(JSON.stringify(newMetadata));
+			console.log("metadata ipfs result", result);
+			
+			// if you get ipfs, overwrite the image_url
+			newTokenURI = "ipfs://" + result.path + "/";
+		} catch (error) {
+			console.log('Error uploading metadata to IPFS: ', error);
+			errorPreparingMint = true;
+			return; 
+		}
```

Comment:
Moving the IPFS upload to a function would be good, used twice already in the same way here